Oppdaterer stegvelger når menypanel endres

martonSkjevelandNav opened the pull request:
Oppdaterer stegvelger om noen av verdiene under familieforhold eller arbeidsgiver/virksomhet endres i menypanel.

Ønsker helst en mer generisk løsning, men føler ikke jeg har god nok kontroll i frontenden til å gjøre dette. Vil fort kreve en gjenskriving av hele stegvelger. Men virker nyttig, så burde vurdere å se på det noen som har litt mer kontroll.

Denne løser iaf kravene til MELOSYS-4449

mar2n committed:
Oppdaterer stegvelger når familieforhold eller arbeidsgiver/virksomhet endres i menypanel

janitamarielle commented:
Jeg får ikke stegene til å oppdatere seg etter endringer i virksomheter 🤔 Jeg må fortsatt gå frem og tilbake i stegene før endringene registreres i virksomhet-steget. Testet med FTRL steg 2.

Commit:
Generisk løsning for å oppdatere stegvelger når menypanel endres

JohanSakshaug commented:
Tok meg den frihet å forsøke å lage en mer generisk løsning her 😄 

I [13f5d7c332ce6c4c365ca05b61f74fb45e35bfd6](https://github.com/navikt/melosys-web/pull/1228/commits/13f5d7c332ce6c4c365ca05b61f74fb45e35bfd6) blir stegvelger oppdatert ved alle endringer i props, men med en debounce. Behandlingsgrunnlag blir også POSTet ved endringer i menypanel, det var nødvendig for å få oppdatert stegvelger.

janitamarielle commented on src/felleskomponenter/stegvelger/Stegvelger.js:
```diff
@@ -465,6 +450,10 @@ class Stegvelger extends Component {
+  debouncedOppdaterAktuelleSteg = Utils._debounce(async (aktivtStegNummer) => {
+    this.oppdaterAktuelleSteg(aktivtStegNummer);
+  }, 300);
+
```
Trenger vi debounce både ved `debouncedOppdaterAktuelleSteg` og ved `debouncedLagreBehandlingsgrunnlag`?

JohanSakshaug replied:
Hvis jeg husker riktig:
debouncedOppdaterAktuelleSteg trenger debounce fordi det er mer enn bare endring av behandlingsgrunnlaget som kan trigge denne. Mener å huske den har en tendens til å bli kalt veldig mange ganger etter hverandre.
debouncedLagreBehandlingsgrunnlag har debounce slik at ikke hver endring i `formValues` skal lagre behandlingsgrunnlag på nytt.

janitamarielle commented:
Når jeg tester lokalt med docker-compose støter jeg på problemer fordi behandlingsgrunnlaget ikke er gyldig. Spesifikt gjelder dette at `Trygdedekninger` er `""`. Burde det være slik at behandlingsgrunnlaget ikke blir lagret før det er gyldig?

JohanSakshaug commented:
> Spesifikt gjelder dette at Trygdedekninger er "". Burde det være slik at behandlingsgrunnlaget ikke blir lagret før det er gyldig?

Blir nok riktigst å hindre at `Trygdedekninger` er `""` 👀

Commit:
Lagring av behandlingsgrunnlag var stuck i uendelig loop.

(antallMaanederINorge er primitiv int i backend, så dersom vi poster antallMaanederINorge: null, vil vi få tilbake antallMaanederINorge: 0. Det trigget ny lagring av behandlingsgrunnlag)

Commit:
test

janitamarielle commented:
Kan se ut som om `overgangsregelbestemmelser` og `ytterligereInformasjon` blir initialisert ved "klikk"(åpner legg-til-barn komponenten) og så fjernet ved henting av behandlingsgrunnlaget 🤔. Dette gjør at behandlingsgrunnlaget blir postet før det har kommet _faktiske_ endringer.

Commit:
Unngå unødvendig oppdatering av redux-state for overgangsregelbestemmelser og ytterligereInformasjon

martonSkjevelandNav reviewed (commented):
Testet at panel oppdateres ok i begge tilfeller:
- Virksomhet-stegene er avhengig av endinger under Arbeidsgiver/virksomhet
- Familie-stegene er avhengig av noen av endringene under Famiieforhold

janitamarielle committed:
Merge branch 'master' of https://github.com/navikt/melosys-web into MELOSYS-4449_refresh-stegvelger-oppdatering-menypanel

Commit:
Merge branch 'master' into MELOSYS-4449_refresh-stegvelger-oppdatering-menypanel

janitamarielle committed:
Merge branch 'MELOSYS-4449_refresh-stegvelger-oppdatering-menypanel' of https://github.com/navikt/melosys-web into MELOSYS-4449_refresh-stegvelger-oppdatering-menypanel

janitamarielle committed:
Trygdedekning blir default undefined istedenfor ""

janitamarielle committed:
Fjerner extra oppdatering av behandlingsgrunnlaget fra vurderingStart-steg, og viser ikke saksbehandling før saksopplysninger er hentet
